packages/graphql-hooks/src/middlewares/apqMiddleware.js

```diff
@@ -1,9 +1,14 @@
-async function sha256(data) {
-  // MOCK
-  return data
-    .split('')
-    .reduce((acc, ch) => acc + ch.charCodeAt(0), '')
-    .substr(0, 8)
+import { Sha256 } from '@aws-crypto/sha256-browser'
+
+export async function sha256(query) {
+  const hash = new Sha256()
+  hash.update(query, 'utf8')
+  const hashBuffer = await hash.digest()
+  // convert buffer to byte array
+  const hashArray = Array.from(hashBuffer)
+  // convert bytes to hex string
+  const hashHex = hashArray.map(b => b.toString(16).padStart(2, '0')).join('')
+  return hashHex
```

Comment:
This might be a bit cleaner, but it's not that big of a deal IMO:

```suggestion
  const hashUint8Array = await hash.digest()

  const hashBuffer = Buffer.from(hashUint8Array)
  return hashBuffer.toString('hex')
```

The `Buffer` approach should also perform better due to Node's internal optimizations. A quick benchmark I've ran showed it's a little bit over 50% faster, with different queries (to avoid any caching) with length `100`.

Reply:
Agreed on this - let's use built-ins where possible, unless there's a compelling reason not to.